### PR TITLE
Fallback to sending an invite as a bot if the regular invite fails

### PR DESCRIPTION
**Pull request by tadzik**
This works around an issue where an invite is sent before the room is
fully configured, and the power level of the actual inviter is not yet
know and set.

Fixes GH-1466.

### CODE REVIEW

File: src/bridge/IrcHandler.ts [hunk 1]
```diff
@@ -406,7 +406,15 @@ export class IrcHandler {
                 virtualMatrixUser.getId()
             ).invite(
                 room.getId(), invitee
-            );
+            ).catch(err => {
+                req.log.warn(
+                    `Failed to invite %s as the inviter user (reason: ${err}),
```
Comment (Half-Shot) [inline src/bridge/IrcHandler.ts:411]: You didn't provide an arg to %s, but we might as well use new template syntax anyway.
```suggestion
                    `Failed to invite ${invitee} as the inviter user (reason: ${err}),
```

File: src/bridge/IrcHandler.ts [hunk 2]
```diff
@@ -406,7 +406,15 @@ export class IrcHandler {
...
+                    inviting as a bot as fallback`
+                );
+                // TODO Note the original inviter in the reason
```
Comment (Half-Shot) [inline src/bridge/IrcHandler.ts:414]: We could just set some room state here tbh, I don't think we should wait for the MSC to arrive :older_man:

**Review by Half-Shot — CHANGES_REQUESTED** [review]

File: src/bridge/IrcHandler.ts (hunk 2 above)
Comment (tadzik) [reply src/bridge/IrcHandler.ts:414]: What could we set in a room state that'll convey the relevant info?

Comment (Half-Shot) [reply src/bridge/IrcHandler.ts:414]: Oh, I mean literally send a m-room-member event with a membership of invite and a reason of whatever.

Comment (tadzik) [reply src/bridge/IrcHandler.ts]: Ah yes, that works. Though currently Synapse seems to just strip the reason anyway, so it doesn't actually show up – but we should at least be future-proof with this :) Done in 2e80b23e.

Comment (Half-Shot) [reply src/bridge/IrcHandler.ts]: :roll_eyes: synaaaaapse. Is there a bug for that? Otherwise, sounds good.

**Review by Half-Shot — APPROVED** [review]

File: src/bridge/IrcHandler.ts (hunk 2 above)
Comment (tadzik) [reply src/bridge/IrcHandler.ts]: (Seems like it does not strip it after all, all works well, so this is all good).

**Pull request merged**